include/kodlab_mjbots_sdk/lcm_subscriber.h

```diff
@@ -1,85 +1,102 @@
 #include "kodlab_mjbots_sdk/abstract_realtime_object.h"
+#include "kodlab_mjbots_sdk/lcm_message_handler.h"
+
+namespace kodlab
+{
 
-namespace kodlab {
-/*!
- * @brief An template class for an lcm subscriber. Subscribes to just one msg of type msg_type.
- * @tparam MessageClass the lcm msg type
+/**
+ * @brief LCM subscriber capable of subscribing to multiple channels
+ * @todo Implement subscription removal (i.e., unsubscribing)
  */
-template<class MessageClass>
-class LcmSubscriber : public AbstractRealtimeObject {
- public:
+class LcmSubscriber : public AbstractRealtimeObject
+{
+public:
+
   /*!
-   * @brief constructor for lcm subscriber
-   * @param realtime_priority the realtime priority, max is 99
-   * @param cpu the cpu for this process
-   * @param channel_name the string for the channel name
+   * @brief Constructor an lcm subscriber
+   * @param realtime_priority realtime priority in range [1, 99]
+   * @param cpu cpu for this process
    */
-  LcmSubscriber(int realtime_priority, int cpu, std::string channel_name);
+  LcmSubscriber(int realtime_priority, int cpu)
+  {
+    cpu_ = cpu;
+    realtime_priority_ = realtime_priority;
+    Start();
+  }
 
-  std::mutex mutex_;         /// Mutex for if m_data
-  bool new_message_ =
-      false; /// True if there is new data, false if data is old, used to prevent user for checking too often
-  MessageClass data_;            /// A copy of the most recent lcm data
- protected:
-  /*!
-   * @brief callback function when msg is received. Copies the message to m_data
-   * @param rbuf unknown, but not used
-   * @param chan channel name
-   * @param msg ptr to the incoming message data
+  /**
+   * @brief Add a new LCM channel to the the LCM object's subscriptions
+   * @tparam Message LCM message type for channel
+   * @param channel_name channel name
+   * @param handler `LcmMessageHandler`-inherited message handler object
+   * @return generated `lcm::Subscription` object
    */
-  void HandleMsg(const lcm::ReceiveBuffer *rbuf,
-                 const std::string &chan,
-                 const MessageClass *msg);
+  template<class Message>
+  lcm::Subscription *AddSubscription(std::string channel_name,
+                                     LcmMessageHandler<Message> &handler)
+  {
+    auto sub = lcm_.subscribe(channel_name,
+                              &LcmMessageHandler<Message>::HandleMessage,
+                              &handler);
+    subs_.emplace(channel_name, sub);
+    return sub;
+  }
 
-  /*!
-   * @brief subscribes to the lcm channel using handle message and handles ctrl c detection
+  /**
+   * @brief Accessor for `lcm::Subscription` objects
+   * @param channel_name channel name
+   * @return `lcm::Subscription` object on corresponding channel
    */
-  void Run() override;
+  [[nodiscard]] const lcm::Subscription *get_subscription(const std::string &channel_name) const
+  {
+    return subs_.at(channel_name);
+  }
 
-  std::string channel_name_;
+protected:
```

Comment:
why protected? Are we expecting to have this class be inherited?

Reply:
Philosophically, I feel like I would default to protected unless there is some explicitly reason to protect the interior member, so that it could be derived if someone wanted to. Is this bad practice?

Reply:
Protected vs private comes into play when you use inheritance with private being the standard. Having it be protected suggests to the reader that this class is expected to be override.

Reply:
Yea I guess the point is intent vs capability, with a leaning toward intent. We don't want us or the user to derive so it should be private and it can change if we decide to do something else in the future. Same reason not every destructor is virtual by default.

Reply:
@jdcaporale has it covered, I left it protected in case the user took it upon themselves to make a derived class, but there was no intention of making the user do this.  Will change to private.

Reply:
Made private in 2130f57a47c8a4f99b5e132f14dc3884a8383f17.